Misc updates to the README (#48)
yuvipanda/jupyterhub-ssh: README.md

   --set sftp.enabled=false
```

If you install JupyterHub SFTP, then it needs access to the home folders. These
home folders are assumed to be exposed via a k8s PVC resource that you should
name via the `sftp.pvc.name` configuration.

If your JupyterHub has been deployed using [the official JupyterHub Helm
chart](https://z2jh.jupyter.org) version 1.1.0 or later, and you have
_configured the official JupyterHub Helm chart_ with `proxy.https.enabled=true`
and `proxy.https.type=letsencrypt`, then you can add the following to to acquire
access to the jupyterhub-ssh and jupyterhub-sftp services via that setup.

```yaml
# Configuration for the official JupyterHub Helm chart to accept traffic via
proxy:
   https:
      enabled: true
      type: letsencrypt
      letsencryptEmail: <my-email-here>

   service:
     # jupyterhub-ssh/sftp integration part 1/3:
     #
     # We must accept traffic to the k8s Service (proxy-public) receiving traffic
     # from the internet. Port 22 is typically used for both SSH and SFTP, but we
     # can't use the same port for both so we use 23.
     #
     extraPorts:
       - name: ssh
         port: 22
         targetPort: ssh
       - name: sftp

>          port: 23

Reviewer: 23 is the port allocated to telnet (https://en.wikipedia.org/wiki/List_of_TCP_and_UDP_port_numbers), so let's use something else? I prefer 2222, since sftp is normally in 22.

>      # from the internet. Port 22 is typically used for both SSH and SFTP, but we
>      # can't use the same port for both so we use 23.
>      #
>      extraPorts:
>        - name: ssh
>          port: 22
>          targetPort: ssh
>        - name: sftp
>          port: 23

Author: True, but 2222 is reserved for `EtherNet/IP implicit messaging for IO data` - isn't it quire arbitrary?

There are plenty of mapping back and forth between ports, I felt it became a bit more clear to use something other than the container's default port. But sure, it is arbitrary, and it is an example!

```suggestion
     # from the internet. Port 22 is typically used for both SSH and SFTP, but we
     # can't use the same port for both so we use 2222 for SFTP in this example.
     #
     extraPorts:
       - name: ssh
         port: 22
         targetPort: ssh
       - name: sftp
         port: 2222
```

Reviewer: While arbitrary, I think 23 for telnet is much more used than 2222. If we run on 23, I guarantee we'll have a million bots trying to brute force us. It'll be the case with 2222 too, but less so

         targetPort: sftp

   traefik:
     # jupyterhub-ssh/sftp integration part 2/3:
     #
     # We must accept traffic arriving to the autohttps pod (traefik) from the
     # proxy-public service. Expose a port and update the NetworkPolicy
     # to tolerate incoming (ingress) traffic on the exposed port.
     #
     extraPorts:
       - name: ssh
         containerPort: 8022

>        - name: sftp
>          containerPort: 8023

Author: ```suggestion
       - name: sftp
         containerPort: 8022
```

>        - name: sftp
>          containerPort: 8023

Author: ```suggestion
       - name: sftp
         containerPort: 2222
```

     networkPolicy:
       allowedIngressPorts: [http, https, ssh]

     # jupyterhub-ssh/sftp integration part 3/3:
     #
     # We must let traefik know it should listen for traffic (traefik entrypoint)
     # and route it (traefik router) onwards to the jupyterhub-ssh k8s Service
     # (traefik service).
     #
     extraStaticConfig:
       entryPoints:
         ssh-entrypoint:
           address: :8022

>          ssh-entrypoint:
>            address: :8023

Author: ```suggestion
         sftp-entrypoint:
           address: :8022
```

>            address: :8023

Author: ```suggestion
           address: :2222
```

     extraDynamicConfig:
       tcp:
         services:
           ssh-service:
             loadBalancer:
               servers:
                 - address: jupyterhub-ssh:22
           sftp-service:
             loadBalancer:
               servers:
                 - address: jupyterhub-sftp:22
         routers:
           ssh-router:

>              entrypoints:
>                - ssh-entrypoint

Author: ```suggestion
             entrypoints: [ssh-entrypoint]
```

             rule: HostSNI(`*`)
             service: ssh-service
           sftp-router:

>              entrypoints:
>                - sftp-entrypoint

Author: ```suggestion
             entrypoints: [sftp-entrypoint]
```
